### PR TITLE
Makefile tweaks

### PR DESCRIPTION
**Pull request by mvdan**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -8,8 +8,8 @@ deps:
 	go get github.com/tools/godep
 
 build: clean deps
-	GOOS=linux GOARCH=amd64 CGO_ENABLED=0 godep go build -a ./...
```
Comment (vishh) [inline Makefile]: This is useful when compiling on OSX. If we switch to docker based builds, we can get rid of this safely.

Comment (mvdan) [inline Makefile]: Oh. Fair enough.

**Comment by mvdan** [comment]
@vishh this makes two flaky tests fail way more often, which I think is a good thing. They are `TestSetSinksStore` and `TestRealCacheData`.

**Comment by mvdan** [comment]
Just fixed `TestRealCacheData`. Left it running in a forever loop for nearly an hour, no test failures.

**Comment by mvdan** [comment]
@vishh the GC bug is a serious regression in the latest stable, perhaps a release is due?

**Comment by vishh** [comment]
@mvdan: Thanks for fixing the race. Which GC bug are you referring to?

**Comment by mvdan** [comment]
There was no race, it was a bug in the garbage collection that I just fixed. That is what I meant with "GC bug". It's a serious regression because it removes all nodes at every `runGC()`.

**Comment by vishh** [comment]
Have you posted a PR?

**Comment by vishh** [comment]
@mvdan: Can you include just the makefile changes in this PR?

**Comment by mvdan** [comment]
Sure.

BTW, the `TestSetSinksStore` flake was introduced in 6d4d6e2251fb42e355. It was definitely not flaky when I made my "changing sinks at runtime" changes.

**Comment by mvdan** [comment]
Mentioning the flake because it now happens almost always thanks to `-race`. In the long run we want `-race` enabled and the flake fixed. In the short run, if we can't fix the flake, we can hold off `-race` for now.

**Comment by vishh** [comment]
We can fix the flake. I can take a look at the failing test.

**Comment by afein** [comment]
regarding the failing test: https://github.com/kubernetes/heapster/pull/478#issuecomment-136442475

**Comment by mvdan** [comment]
All green now thanks to #510, merging

**Pull request merged**